icmplib/sockets.py

```diff
@@ -146,6 +154,31 @@ def _set_traffic_class(self, traffic_class):
         '''
         raise NotImplementedError
 
+    def _bind_to_device(self, interface):
+        '''
+        Bind the socket to a specific network interface.
+
+        Only available on Linux. Ignored on macOS and Windows.
+
+        '''
+        # Not available on Windows or macOS
+        if PLATFORM_WINDOWS or PLATFORM_MACOS:
+            return
+
+        # Linux uses SO_BINDTODEVICE
+        # SO_BINDTODEVICE requires CAP_NET_RAW capability on Linux
+        # Interface name must be encoded to bytes
+        # Try to get SO_BINDTODEVICE constant, fallback to 25 for compatibility
+        SO_BINDTODEVICE = getattr(socket, "SO_BINDTODEVICE", 25)
+        try:
+            self._sock.setsockopt(
+                socket.SOL_SOCKET,
+                SO_BINDTODEVICE,
+                interface.encode("ascii"),
+            )
+        except (OSError) as err:
```

Comment:
Nit: unnecessary parantheses
```suggestion
        except OSError as err:
```